src/mumble/CoreAudio.mm

```diff
@@ -1108,3 +1115,20 @@ static void LogAUStreamDescription(AudioUnit au) {
 		qWarning("CoreAudioOutput: Unexpected property changed event received.");
 	}
 }
+
+OSStatus CoreAudioOutput::deviceChange(AudioObjectID inObjectID, UInt32 inNumberAddresses,
+                                       const AudioObjectPropertyAddress inAddresses[], void *udata) {
+	Q_UNUSED(inObjectID);
+	Q_UNUSED(inNumberAddresses);
+	Q_UNUSED(inAddresses);
+
+	CoreAudioOutput *o = reinterpret_cast< CoreAudioOutput * >(udata);
+	if (!o->bRunning) return noErr;
+
+	qWarning("CoreAudioOutput: Output device change detected. Restarting AudioOutput.");
+	o->stop();
+	Audio::stopOutput();
+	Audio::startOutput();
```

Comment:
Did you check whether explicitly calling `o->stop()` is required? It seems like `Audio::stopOutput()` should already take care of that :thinking:

Reply:
I didn't actually check that, no - I'll see what happens if it's removed

Reply:
You're right, it wasn't required - removed 👍